IM_calculation/scripts/calculate_rrups_single.py

```diff
@@ -66,43 +71,64 @@ def write_source_2_site_dists(
         "will be calculated for all stations in the fd_station_file "
         "and the station_file",
     )
+    parser.add_argument(
+        "-b",
+        "--backarc",
+        action="store_true",
+        help="Add a crude definition of back-arc to the output file. 1 if on back-arc, 0 if on forearc",
+    )
 
     args = parser.parse_args()
 
     # Load the stations
     station_df = load_station_file(args.station_file)
 
-    # Get the location for all station in the fd station file
+    # Matches the station file with the specified stations or the fd_stat_list. If neither is specified, use the whole list
     if args.fd_station_file:
```

Comment:
Could this whole block be simplified?
```python
station_mask = np.ones(station_df.shape[0], dtype=bool)
if args.fd_station_file:
    fd_stations_df = load_station_file(args.fd_station_file)
    station_mask = np.isin(station_df.index.values, fd_stations_df)
elif args.stations:
    station_mask = np.isin(station_df.index.values, args.stations)

filtered_station_np = station_df.loc[station_mask].values
stats_2_calc = station_df.loc[station_mask].index.values

# Or just use the dataframe directly?
station_df = station_df.loc[station_mask]
```
Something along those lines?

Reply:
Sweet. The mask is a good idea. That makes it a wee bit simpler. 

I'd have preferred to use the df everywhere but I don't want to dig around and change too much as it's just a stylistic change rather than a readability change etc... Someone originally decided to use numpy for this script.